src/Model/Item.php

```diff
@@ -4198,12 +4199,16 @@ public static function fetchByLink(string $uri, int $uid = 0, int $completion =
 		}
 
 		try {
-			$curlResult = DI::httpClient()->head($uri, [HttpClientOptions::ACCEPT_CONTENT => HttpClientAccept::JSON_AS, HttpClientOptions::REQUEST => HttpClientRequest::ACTIVITYPUB]);
-			if (!HTTPSignature::isValidContentType($curlResult->getContentType(), $uri) && (current(explode(';', $curlResult->getContentType())) == 'application/json')) {
+			if (empty($mimetype)) {
+				$curlResult = DI::httpClient()->head($uri, [HttpClientOptions::ACCEPT_CONTENT => HttpClientAccept::JSON_AS, HttpClientOptions::REQUEST => HttpClientRequest::ACTIVITYPUB]);
+				$mimetype = $curlResult->getContentType();
+			}
+			if (!HTTPSignature::isValidContentType($mimetype, $uri) && (current(explode(';', $mimetype)) == 'application/json')) {
 				// Issue 14126: Workaround for Mastodon servers that return "application/json" on a "head" request.
 				$curlResult = HTTPSignature::fetchRaw($uri, $uid);
```

Comment:
`$curlResult` can be undefined. You can try to get an early return to avoid this type of errors.

Reply:
What do you mean with "early returns"?

Reply:
> And where do you see a condition that could lead to an undefined `$curlResult`?

I'm sorry, I must have misread this line on my small display because of the check warnings. 😅

![grafik](https://github.com/user-attachments/assets/a756cfe0-c189-44c3-be90-fdb1d6abf895)

Instead `$curlResult` can be `null` if `HTTPSignature::fetchRaw()` returns noting. This will lead to an fatal error in the next line.

> What do you mean with "early returns"?

"Early return" aka "flattening conditions" aka "bringing the code to the left" is a technique to returning from a function as soon as possible when certain conditions are met, rather than nesting multiple if-else statements. This can improve readability and maintainability in functions or methods.

Please see the examples at the end of this article to gasp the idea: https://dev.to/bespoyasov/refactoring-tools-early-return-for-flatter-conditions-d1p

Reply:
We have to encapsulate both HTTP commands with `try ... except`. I will have a look at the proposal later this day.

Reply:
That's another nice effect of early returns: You are seeing your edge cases much more clearer along the happy path. Now If you are finding yourself writing some things multiple times inside the same method it is a strong indicator that this code could be extract into a separate method and called multiple times. This is part of the WET (Write Everything Twice) paradigm.

Reply:
I rearranged the code to make it clearer for me.